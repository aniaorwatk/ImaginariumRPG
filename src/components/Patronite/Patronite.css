.patronite {
    display: flex;
    align-items: center;
     gap: 13px;
    width: 342px;
    height: 66px;
    border-bottom: 0.75px solid rgba(255, 255, 255, 0.7);
    border-top: 0.75px solid rgba(255, 255, 255, 0.7);
    font-size: 16px;
}

.patronite__email {
    font-weight: 600;
    color: rgba(255, 255, 255, 0.8);
}
