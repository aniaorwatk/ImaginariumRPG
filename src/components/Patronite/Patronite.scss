@import "./../scss/Main";

.patronite {
    display: flex;
    align-items: center;
    justify-content: center;
    gap: 13px;
    width: 342px;
    height: 66px;
    border-bottom: $color-white-trans07;
    border-top: $color-white-trans07;
    font-size: 16px;
    color: $color-white-trans08; 
}

.patronite__email {
    font-weight: 600;
    text-decoration: none;
    color: $color-white;
    cursor: pointer;
}

.patronite__email:hover {
    color:$color-white-trans07;
}

.patronite_logoEmail {
    width: 0;
}

.patronite__link{
    text-decoration: none;
    color:$color-white;
    background-color: $color-blue;
    cursor: pointer;
}

@media only screen and (min-width: $breakpointM) {
    .patronite {
        width: 460px;
        height: 66px;
        margin: 36px 0;
    }
}

@media only screen and (min-width: $breakpointL) {
    .patronite {
        width: 912px;
    }

    .patronite__logo {
        width: 167px;
        height: 53.27px;
        z-index:2;
    }

    .patronite__email{
       font-size: 20px;
        line-height: 25px;
    }

    .patronite_logoEmail {
        width: 24px;
    }
}
