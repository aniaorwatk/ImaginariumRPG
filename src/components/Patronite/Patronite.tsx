import { useEffect, useState } from "react";
import { getAction, IDataType, URL_IMAGINARIUM } from "../../API/APIGet";
import { linkPatronite } from "../../links";
import { LogoMail } from "../Logo/LogoMail";
import { LogoPatronite } from "../Logo/LogoPatronite";
import "./Patronite.css";

const Patronite = () => {
    const [patronite, setPatronite] = useState("")

    useEffect(() => {
        getAction(URL_IMAGINARIUM, (data: IDataType) => {
            setPatronite(data.patronite_email)
        })
    }, [patronite]);

    return (
        <div>
            {patronite ? <div className="patronite">
                <>
                    <a href={linkPatronite} target="_blank">
                        <LogoPatronite className="patronite__logo" />
                    </a>
                    |
                    <LogoMail className="patronite_logoEmail" />
                    <a href={`mailto:${patronite}`} target="_blank" className="patronite__email">{patronite}</a>
                </>
            </div>
                :
                <p>Loading...</p>
            }
        </div>
    )
}

export default Patronite
