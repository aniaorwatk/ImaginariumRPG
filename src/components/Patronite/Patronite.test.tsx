import { screen, render } from "@testing-library/react";
import Patronite from "./Patronite";

describe('Patronite Component', () => {
    it('Should render', async () => {
        const { asFragment } = render(<Patronite />);

        await screen.findAllByText('imagrpg@gmail.com');

        expect(asFragment()).toMatchSnapshot();
    })
})
