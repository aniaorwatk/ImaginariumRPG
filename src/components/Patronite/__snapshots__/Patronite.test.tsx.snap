// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Patronite Component Should render 1`] = `
<DocumentFragment>
  <div>
    <div
      class="patronite"
    >
      <a
        href="https://patronite.pl/imaginariumrpg"
        rel="noopener noreferrer"
        target="_blank"
      >
        <svg
          class="patronite__logo"
          title="Logo Patronite"
        >
          patronite.svg
        </svg>
      </a>
      |
      <svg
        class="patronite_logoEmail"
      >
        mail.svg
      </svg>
      <a
        class="patronite__email"
        href="mailto:imagrpg@gmail.com"
        rel="noopener noreferrer"
        target="_blank"
      >
        imagrpg@gmail.com
      </a>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Patronite Should render 1`] = `
<DocumentFragment>
  <div>
    <div
      class="patronite"
    >
      <svg
        class="patronite__logo"
      >
        patronite.svg
      </svg>
      |
      <svg
        class="patronite_logoEmail"
      >
        mail.svg
      </svg>
      <p
        class="patronite__email"
      >
        imagrpg@gmail.com
      </p>
    </div>
  </div>
</DocumentFragment>
`;
